L1Trigger/L1TTrackMatch/plugins/L1TkHTMissEmulatorProducer.cc

```diff
@@ -0,0 +1,286 @@
+    << "====CUT AP_INTS====\n" << "minpt: "<< jetMinPt_ << " maxeta: " << jetMaxEta_ << " minNtracksHighPt: " << minNtracksHighPt_ << " minNtracksLowPt: " << minNtracksLowPt_ << "\n"
+    << "====CUT AP_INTS TO FLOATS====\n" << "minpt: "<< (float) jetMinPt_ * l1tmhtemu::kStepPt << " maxeta: " << (float) jetMaxEta_ * l1tmhtemu::kStepEta << " minNtracksHighPt: " << (int) minNtracksHighPt_ << " minNtracksLowPt: " << (int) minNtracksLowPt_ <<"\n"
+    << "-------------------------------------------------------------------------\n";
+  }
+
+}
+
+L1TkHTMissEmulatorProducer::~L1TkHTMissEmulatorProducer() {}
+
+void L1TkHTMissEmulatorProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  using namespace edm;
+  std::unique_ptr<std::vector<l1t::EtSum>> MHTCollection(new std::vector<l1t::EtSum>(0));
+
+  // L1 track-trigger jets
+  edm::Handle<TkJetWordCollection> L1TkJetsHandle;
+  iEvent.getByToken(jetToken_, L1TkJetsHandle);
+  std::vector<TkJetWord>::const_iterator jetIter;
+
+  if (!L1TkJetsHandle.isValid() && !displaced_) {
+    LogError("TkHTMissEmulatorProducer") << "\nWarning: TkJetCollection not found in the event. Exit\n";
+    return;
+  }
+
+  if (!L1TkJetsHandle.isValid() && displaced_) {
+    LogError("TkHTMissEmulatorProducer") << "\nWarning: TkJetExtendedCollection not found in the event. Exit\n";
+    return;
+  }
+
+  int cosLUTbins = floor(l1tmhtemu::kMaxCosLUTPhi / l1tmhtemu::kStepPhi);
+
+  // floats used for debugging
+  float sumPx_ = 0;
+  float sumPy_ = 0;
+  float HT_ = 0;
+
+  float sumPxtemp_ = 0;
+  float sumPytemp_ = 0;
+  float HTtemp_ = 0;
+
+  l1tmhtemu::Et_t sumPx = 0;
+  l1tmhtemu::Et_t sumPy = 0;
+  l1tmhtemu::Et_t HT = 0;
+
+  // loop over jets
+  int jetn = 0;
+  int jetnpasscuts = 0;
+  for (jetIter = L1TkJetsHandle->begin(); jetIter != L1TkJetsHandle->end(); ++jetIter) {
+
+    // floats used for debugging
+    float tmp_jet_px_ = jetIter->pt()*cos(jetIter->glbphi());
+    float tmp_jet_py_ = jetIter->pt()*sin(jetIter->glbphi());
+    float tmp_jet_et_ = jetIter->pt(); // FIXME Get Et from the emulated jets
+    float tmp_jet_pt_ = jetIter->pt();
+    
+
+    l1tmhtemu::pt_t tmp_jet_pt = l1tmhtemu::digitizeSignedValue<l1tmhtemu::pt_t>(jetIter->pt(), l1tmhtemu::kInternalPtWidth, l1tmhtemu::kStepPt);
+    l1tmhtemu::eta_t tmp_jet_eta = l1tmhtemu::digitizeSignedValue<l1tmhtemu::eta_t>(jetIter->glbeta(), l1tmhtemu::kInternalEtaWidth, l1tmhtemu::kStepEta);
+    l1tmhtemu::phi_t tmp_jet_phi = l1tmhtemu::digitizeSignedValue<l1tmhtemu::phi_t>(jetIter->glbphi(), l1tmhtemu::kInternalPhiWidth, l1tmhtemu::kStepPhi);
+    l1tmhtemu::ntracks_t tmp_jet_nt = l1tmhtemu::ntracks_t(jetIter->nt());
+
+    l1tmhtemu::phi_t tmp_jet_cos_phi = l1tmhtemu::phi_t(-999);
+    l1tmhtemu::phi_t tmp_jet_sin_phi = l1tmhtemu::phi_t(-999);
+
+
+    if(tmp_jet_phi >= 0){
+      tmp_jet_cos_phi = cosLUT_[tmp_jet_phi];
+
+      if(cosLUTbins/2 + 1 - tmp_jet_phi >= 0){
+	tmp_jet_sin_phi = cosLUT_[cosLUTbins/2 + 1 - tmp_jet_phi];
```

Comment:
Your spacing/alignment seems off here. Please make sure the style/spacing is uniform throughout the file.

Reply:
should be fixed in the latest commit

Reply:
Are you sure? It certainly still looks a little off to me on GitHub.

Reply:
hopefully, looks okay now?

Reply:
Yes, it does.